agent/php_internal_instrument.c

```diff
@@ -1527,10 +1560,63 @@ NR_INNER_WRAPPER(mysqli_stmt_prepare) {
  */
 NR_INNER_WRAPPER(memcache_function) {
   nr_php_instrument_datastore_operation_call(nr_wrapper, NR_DATASTORE_MEMCACHE,
-                                             nr_wrapper->extra, NULL,
+                                             nr_wrapper->extra, NULL, false,
                                              INTERNAL_FUNCTION_PARAM_PASSTHRU);
 }
 
+NR_INNER_WRAPPER(memcached_connect_function) {
+  char* host = NULL;
+  nr_string_len_t host_len = 0;
+  zend_long port = 0;
+  zend_long weight = 0;
+  nr_datastore_instance_t* instance = NULL;
+
+  if (SUCCESS
+      == zend_parse_parameters_ex(
+          ZEND_PARSE_PARAMS_QUIET, ZEND_NUM_ARGS() TSRMLS_CC, "s|ll", &host,
+          &host_len, &port, &weight)) {
+    if (NULL != host) {
+      instance = nr_php_memcached_create_datastore_instance(host, port);
+    }
+  }
+  nr_php_instrument_datastore_operation_call(nr_wrapper, NR_DATASTORE_MEMCACHE,
+                                             NULL, instance, true,
+                                             INTERNAL_FUNCTION_PARAM_PASSTHRU);
```

Comment:
Is there a point in calling `nr_php_redis_save_datastore_instance` when `instance` is `NULL`?

Reply:
No, but `instance` should never be `NULL` and is still eventually checked. `instance` is a string created from `host`, which we check for `NULL` above. So `instance` should only be `NULL` if an allocation failed. Given that the code is still safe if that happens, I am disinclined to add a check.

Reply:
this call was moved inside the `if`'s scope in [155bc5f](https://github.com/newrelic/newrelic-php-agent/pull/958/commits/155bc5ff645557bf0d001f691b2982fbf51fcf48)

Reply:
This comment is obsolete as of 3ed5bc54bb0d90bc792ed32cd650cfb87c7de972